### ISSUE TITLE
Razor does not strip escaped @ in CSS blocks from virtual documents

### ISSUE THREAD

**Issue by jimmylewis** [dotnet/razor-compiler#306]

**Comment by NTaylorMullen** [comment]
Investigated and this is due to the Razor compilers `[GetHtmlDocument](https://github.com/dotnet/razor-compiler/blob/a4df79c2bcf7d6e79cf870f84cf9fe3170c83fcf/src/Microsoft.AspNetCore.Razor.Language/src/RazorCodeDocumentExtensions.cs#L121)` not rendering the single `@` when a `@@` is mentioned for the HTML variant of the output. Looks like the `[RazorHtmlWriter](https://github.com/dotnet/razor-compiler/blob/a4df79c2bcf7d6e79cf870f84cf9fe3170c83fcf/src/Microsoft.AspNetCore.Razor.Language/src/RazorHtmlWriter.cs)` doesn't handle the `@@` case properly.

Here's the generated HTML, note the `@@` in front of font-face instead of "whitespace" `@`:
![image](https://user-images.githubusercontent.com/2008729/180881722-5b1d0ddd-a1f2-4d47-8e16-c9a21ceae4c1.png)

**Comment by oliverholliday** [comment]
Also experiencing this - the only warnings for our entire project are false-positives from inline CSS parsing. The problem seems to exist both ways round - in one direction as in the OP where you want to escape to get a literal @, but also in the other case where you're referencing a variable inline:

```
<style type="text/css">
:root {
    --header-height: @(HeaderHeight)px;
}
</style>
@code {
    private int HeaderHeight { get; set; } = 84;
}
```
> CSS039 Unexpected character sequence in property value.
